test/test_um2netcdf.py

```diff
@@ -70,46 +70,28 @@ def mule_vars(z_sea_rho_data, z_sea_theta_data):
     return um2nc.MuleVars(um2nc.GRID_NEW_DYNAMICS, d_lat, d_lon, z_sea_rho_data, z_sea_theta_data)
 
 
-def set_default_attrs(cube, item_code: int, var_name: str):
-    """Add subset of default attributes to flesh out cube like objects."""
-    cube.__dict__.update({"item_code": item_code,
-                          "var_name": var_name,
-                          "long_name": "",
-                          "coord": {"latitude": 0.0,  # TODO: real val = ?
-                                    "longitude": 0.0},  # TODO: real val
-                          "cell_methods": [],
-                          "data": None,
-                          })
-
-    section, item = um2nc.to_stash_code(item_code)
-    cube.attributes = {um2nc.STASH: DummyStash(section, item)}
-
-
 @pytest.fixture
-def air_temp_cube():
-    # data copied from aiihca.paa1jan.subset file
-    name = "air_temperature"
-    m_air_temp = mock.NonCallableMagicMock(spec=iris.cube.Cube, name=name)
-    set_default_attrs(m_air_temp, 30204, name)
-    return m_air_temp
+def air_temp_cube(lat_river_coord, lon_river_coord):
+    # details copied from aiihca.paa1jan.subset file
+    air_temp = DummyCube(30204, "air_temperature",
+                         coords=[lat_river_coord, lon_river_coord])
+    return air_temp
 
 
 @pytest.fixture
-def precipitation_flux_cube():
+def precipitation_flux_cube(lat_river_coord, lon_river_coord):
     # copied from aiihca.paa1jan.subset file
```

Comment:
This one's bordering on pedantry so it's not a problem if you leave these as is, but the `lat_river` and `lon_river` coordinates aren't used for very many variables, and the `air_temp_cube`, `precipitation_flux_cube`, `geopotential_cube`, heaviside_uv_cube`, `heaviside_t_cube`, `ta_plev_cube` are more likely to use the `lon/lat_standard`, or `lon_u/lat_v` coordinates.

If you think it's worth changing these fixtures, I can look into some model output to see which coordinates these variables use, otherwise it's probably ok to leave as is!

Reply:
The tests are a form of documentation & should aim to reflect real data.

If it only takes a short while to dig out the grid types, swapping fixtures is a matter of a few renames. Otherwise, if it's fiddly/time consuming, I can add a comment to indicate the data is invented & not to copy it. Comments can go unread, so it's not the "best" option.

Reply:
It should be pretty easy to get the grids! I'll do that now

Reply:
I've looked at some output for ESM1.5, and I think we can use:
`heaviside_uv_cube`, `ta_plev_cube`, `air_temp_cube`, and `geopotential_cube` all would use `lat_v_nd` and `lon_u_nd` as their coordinates.

`precipitation_flux_cube` would use `lat_standard_nd` and `lon_standard_nd`.

The same variables from CM2 would use different coordinates, but if we are happy to use examples from ESM1.5 then the above should be ok.

Our ESM1.5 configs don't save the `heaviside_t` variable, but for CM2 `heaviside_t_cube` would use the `lat_standard_eg` and `lon_standard_eg`.

I noticed a couple of small typos in the stash codes. I think `geopotential_cube`  should have stash code `30207` instead of `30297`, and `ta_plev`'s stash code should be `30204` instead of `30294`. Changing the `ta_plev_cube`'s stash code makes it a duplicate of `air_temp_cube` though (other than the name), and so we may be able to remove one of the two.

Reply:
Several fixes have gone in:
* 3a774ed2 swaps to the other coordinate fixtures & fixes the item codes
* 8cf02c84 removes `air_temp_cube` & replaces with `ta_plev` --> tests still pass.
* Additional code ordering, moving test fixtures near the module's head (bring code block order closer to top to bottom "reading" order)

Reply:
Apologies, my comment about the typo's in the STASH codes was incorrect!

CM2 does in fact output a `geopotential height` variable with stash code `30297`. The difference between CM2's geopotential height with  stash code `30297` vs ESM1.5's geopotential height with stash code `30207` is the horizontal grid they use. I.e:
- CM2's geopotential height cube with code `30297` would use `lat_standard_eg_coord` and `lon_standard_eg_coord`. [This variable requires the `heaviside_t` cube for masking.](https://github.com/ACCESS-NRI/um2nc-standalone/blob/223c50b5d738d5f455d0e16a9592f8fda81d678f/umpost/um2netcdf.py#L697-L699)
- ESM1.5's geopotential height cube with code `30207` uses `lat_v_nd_coord` and `lon_u_eg_coord`. [This cube requires the `heaviside_uv` cube for masking.](https://github.com/ACCESS-NRI/um2nc-standalone/blob/223c50b5d738d5f455d0e16a9592f8fda81d678f/umpost/um2netcdf.py#L687-L689)


Similar for the `air_temp_cube` vs `ta_plev_cube`:
- CM2 outputs an air temperature cube with stash code `30294` and coordinates `lat_standard_eg_coord`, `lon_standard_eg_coord`. [This cube requires the `heaviside_t` cube for masking.](https://github.com/ACCESS-NRI/um2nc-standalone/blob/223c50b5d738d5f455d0e16a9592f8fda81d678f/umpost/um2netcdf.py#L697-L699)
- ESM1.5 outputs an air temperature cube with code `30204` with coordinates `lat_v_nd_coord` and `lon_u_eg_coord`. [This cube requires the `heaviside_uv` cube for masking](https://github.com/ACCESS-NRI/um2nc-standalone/blob/223c50b5d738d5f455d0e16a9592f8fda81d678f/umpost/um2netcdf.py#L687-L689).

--- 

Because they require different masking cubes (`heaviside_t` vs `heaviside_uv`), the choice of the stash codes in the fixtures impacts the code executed by the `test_process...` tests. https://github.com/ACCESS-NRI/um2nc-standalone/blob/223c50b5d738d5f455d0e16a9592f8fda81d678f/test/test_um2netcdf.py#L181-L296

Based on this, I think it makes sense to:
- Keep `ta_plev_cube` with stash code `30204`, (which requires the `heaviside_uv_cube` mask) as the only air temperature cube. 
- Revert `geopotential_height_cube` to use stash code `30297`, so that it requires the `heaviside_t_cube`.

Which will make sure both the `need_heaviside_uv` and `need_heaviside_t` branches are tested in the process/filtering tests.

Again sorry about this! Let me know if these suggestions sound ok.

Reply:
All good, mistakes happen. This highlights problems with the stash code module containing duplicate names under different item codes --> it's a source of confusion.

I've reverted the `geo_potential_cube` to its original `item_code` & commented on the cubes in a84ba48f.

Reply:
Awesome, those changes look good.

Are you happy to also swap the coordinates on the `geopotential_cube` to `lat_standard_eg_coord` and `lon_standard_eg_coord`? (My mistake in the earlier comment too).

Reply:
Yep, the change is in 8b244b1.

Reply:
Awesome, thanks!